.github/workflows/rust_checks.yml

```
@@ -0,0 +1,125 @@
+name: Rust Static checks
+
+on:
+  pull_request:
+    paths:
+      - .github/workflows/rust_checks.yml
+      - Cargo.lock
+      - Cargo.toml
```

We can just run this on any cargo.toml change in the repo.

```suggestion
      - '**/Cargo.toml'
```